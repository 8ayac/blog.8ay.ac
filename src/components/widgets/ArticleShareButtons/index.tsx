import { config } from '@/site.config';
import { Facebook } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Facebook';
import { Hatena } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Hatena';
import { Line } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Line';
import { Pocket } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Pocket';
import { Reddit } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Reddit';
import { Twitter } from '@/src/components/widgets/ArticleShareButtons/CustomShareButton/Twitter';
import { getArticlePagePath } from '@/src/shared/utils';
import { Article } from '@/src/types';
import styled from '@emotion/styled';
import React from 'react';
import urljoin from 'url-join';

const S = {
  ComponentWrapperSection: styled.section``,

  ButtonListWrapperFlexDiv: styled.div`
    display: flex;
  `,

  ButtonWrapperDiv: styled.div`
    margin-right: 1rem;
    margin-bottom: 0.5rem;
    filter: drop-shadow(
      2px 2px 1px
        ${(props) =>
          props.theme.color.articleShareButtons.buttonWrapperDiv.shadow}
    );

    :hover {
      filter: brightness(120%);
    }
  `,
};

export const ArticleShareButtons: React.FC<{
  article: Article;
  size?: number;
}> = ({ article, size = 32 }) => {
  const shareUrl = urljoin(config.site.rootUrl, getArticlePagePath(article));
  const shareTitle = article.attributes.title;
  const shareButtons = (buttonSize: number) => [
    <Facebook key="facebook" shareUrl={shareUrl} size={buttonSize} />,
    <Twitter
      key="twitter"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Hatena
      key="hatena"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Line
      key="line"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Reddit
      key="reddit"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
    <Pocket
      key="pocket"
      shareUrl={shareUrl}
      shareTitle={shareTitle}
      size={buttonSize}
    />,
  ];

  return (
    <>
      <S.ComponentWrapperSection>
        <S.ButtonListWrapperFlexDiv>
          {shareButtons(size).map((shareButton) => (
            <S.ButtonWrapperDiv key={shareButton.key}>
              {shareButton}
            </S.ButtonWrapperDiv>
          ))}
        </S.ButtonListWrapperFlexDiv>
      </S.ComponentWrapperSection>
    </>
  );
};
