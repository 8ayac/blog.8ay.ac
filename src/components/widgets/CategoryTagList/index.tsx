import { CategoryTag } from '@/src/components/elements/CategoryTag';
import { FlexContainerDiv } from '@/src/shared/styles/abstractStyledComponents';
import styled from '@emotion/styled';
import { faTags } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';

const S = {
  ComponentWrapperDiv: styled(FlexContainerDiv)`
    flex-flow: row wrap;

    > * {
      margin-right: 0.7rem;
      margin-bottom: 0.7rem;
    }
  `,

  PreIconWrapperDiv: styled(FlexContainerDiv)`
    align-items: center;
    color: ${(props) => props.theme.color.text.primaryLight};
  `,
};

export const CategoryTagList: React.FC<{
  tags: string[];
  withPreIcon?: boolean;
}> = ({ tags, withPreIcon = false }) => (
  <>
    <S.ComponentWrapperDiv>
      {withPreIcon && (
        <S.PreIconWrapperDiv>
          <FontAwesomeIcon icon={faTags} />
        </S.PreIconWrapperDiv>
      )}
      {[...new Set(tags)].sort().map((tag) => (
        <CategoryTag key={tag} name={tag} />
      ))}
    </S.ComponentWrapperDiv>
  </>
);
