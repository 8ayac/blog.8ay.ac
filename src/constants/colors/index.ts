export * from './blue';
export * from './green';
export * from './yellow';
