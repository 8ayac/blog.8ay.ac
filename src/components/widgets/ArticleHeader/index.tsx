import { BodyHeader } from '@/src/components/elements/BodyHeader';
import { DateTime } from '@/src/components/elements/DateTime';
import { CategoryTagList } from '@/src/components/widgets/CategoryTagList';
import { getArticlePagePath } from '@/src/shared/utils';
import { Article } from '@/src/types';
import styled from '@emotion/styled';
import React from 'react';

const S = {
  DateTimeWrapperDiv: styled.div`
    display: flex;
    flex-flow: row wrap;
    align-items: center;
    margin-bottom: 1rem;

    > * {
      margin-right: 1rem;
    }
  `,

  TagListWrapperDiv: styled.div`
    margin-top: 2rem;
    margin-bottom: 0.5rem;
  `,
};

export const ArticleHeader: React.FC<{ article: Article }> = ({ article }) => {
  const date = (
    <S.DateTimeWrapperDiv>
      <DateTime
        date={article.attributes.publishedAt}
        description="Published At"
      />
      <DateTime date={article.attributes.updatedAt} description="Updated At" />
    </S.DateTimeWrapperDiv>
  );
  const tags = (
    <S.TagListWrapperDiv>
      <CategoryTagList tags={article.attributes.tags} withPreIcon />
    </S.TagListWrapperDiv>
  );

  return (
    <BodyHeader
      title={article.attributes.title}
      linkTo={getArticlePagePath(article)}
      beforeTitle={date}
      afterTitle={tags}
    />
  );
};
