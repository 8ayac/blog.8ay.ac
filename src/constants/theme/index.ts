import * as colors from '@/src/constants/colors';

export const theme = {
  color: {
    blue: colors.blue,
    green: colors.green,
    yellow: colors.yellow,

    primary: colors.blue.base,
    primaryLight: colors.blue.light,
    primaryDark: colors.blue.dark,

    text: {
      primary: 'hsla(0, 0%, 15%)',
      primaryLight: 'hsla(0, 0%, 40%)',
      secondary: colors.blue.dark2,
      accent: colors.green.dark,
    },
  },
};
