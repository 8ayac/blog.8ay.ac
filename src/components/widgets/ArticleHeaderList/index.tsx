import { ArticleHeader } from '@/src/components/widgets/ArticleHeader';
import { Article } from '@/src/types';

import styled from '@emotion/styled';
import React from 'react';

const S = {
  ComponentWrapperDiv: styled.div`
    display: grid;
    grid-gap: 5%;
    margin: 0 auto;
  `,

  ArticleHeaderWrapperDiv: styled.div`
    h1 {
      font-size: 2.3rem;
    }
  `,
};

export const ArticleHeaderList: React.FC<{ articles: Article[] }> = ({
  articles,
}) => (
  <>
    <S.ComponentWrapperDiv>
      <S.ArticleHeaderWrapperDiv>
        {articles
          .sort(
            (a, b) =>
              new Date(a.publishedAt).getTime() -
              new Date(b.publishedAt).getTime()
          )
          .reverse()
          .map((article) => (
            <ArticleHeader key={article.attributes.id} article={article} />
          ))}
      </S.ArticleHeaderWrapperDiv>
    </S.ComponentWrapperDiv>
  </>
);
