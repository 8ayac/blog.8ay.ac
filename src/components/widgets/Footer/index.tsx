import { theme } from '@/src/constants/theme';
import styled from '@emotion/styled';
import React from 'react';

const S = {
  Footer: styled.footer`
    padding: 3rem 0;
    font-size: 1.5rem;
    text-align: center;
  `,

  CopyRightParagraph: styled.p``,

  AboutGAParagraph: styled.p`
    color: ${theme.color.text.primaryLight};
  `,
};

export const Footer: React.FC = () => (
  <S.Footer>
    <S.CopyRightParagraph>
      Â© 2021 blog.8ay.ac All Rights Reserved.
    </S.CopyRightParagraph>
    <S.AboutGAParagraph>
      This website uses Google Analytics, a web analytics service by Google.
      Google Analytics uses cookies, which are placed on your computer to help
      the website analyze how you use the site. You can prevent the the
      collection of your data by disabling cookies in your browser.
    </S.AboutGAParagraph>
  </S.Footer>
);
