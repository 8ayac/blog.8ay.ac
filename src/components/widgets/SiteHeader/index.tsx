import { BlogTitle } from '@/src/components/elements/BlogTitle';
import { SocialLinkWithIcon } from '@/src/components/elements/SocialLinkWithIcon';
import { BLOG_SUBTITLE, BLOG_TITLE } from '@/src/constants/site';
import { theme } from '@/src/constants/theme';
import { FlexContainer } from '@/src/shared/styles/abstractStyledComponents';
import styled from '@emotion/styled';
import { faTwitter } from '@fortawesome/free-brands-svg-icons';
import React from 'react';

const S = {
  Header: styled.header`
    position: sticky;
    top: 0;
    z-index: 100;
    height: 3.7rem;
    padding: 0.5rem 1rem;
    background-color: ${theme.color.green.dark2};
    border-bottom: 1px solid ${theme.color.green.light};
  `,

  HeaderInner: styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 100%;
    color: white;
  `,

  TitleWrapper: styled(FlexContainer)`
    align-items: center;
  `,

  NavWrapper: styled(FlexContainer)`
    align-items: center;
  `,
};

export const SiteHeader: React.FC = () => (
  <S.Header>
    <S.HeaderInner>
      <S.TitleWrapper>
        <BlogTitle title={BLOG_TITLE} subtitle={BLOG_SUBTITLE} />
      </S.TitleWrapper>

      <S.NavWrapper>
        <SocialLinkWithIcon
          url="http://twitter.com/8ayac"
          description="8ayac"
          icon={faTwitter}
        />
      </S.NavWrapper>
    </S.HeaderInner>
  </S.Header>
);
