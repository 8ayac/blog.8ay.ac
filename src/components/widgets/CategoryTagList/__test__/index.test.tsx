import { CategoryTagList } from '@/src/components/widgets/CategoryTagList';
import { theme } from '@/src/constants/theme';
import { ThemeProvider } from '@emotion/react';
import { mount } from 'enzyme';
import React from 'react';

describe('CategoryTagList', () => {
  const testTags = ['A', 'B', 'C', 'DUPLICATE', 'DUPLICATE'];

  describe('is rendered correctly', () => {
    describe('to match the snapshot', () => {
      test('with preIcon', () => {
        const wrapper = mount(
          <ThemeProvider theme={theme}>
            <CategoryTagList tags={testTags} withPreIcon />
          </ThemeProvider>
        );
        expect(wrapper).toMatchSnapshot();
      });

      test('without preIcon (default)', () => {
        const wrapper = mount(
          <ThemeProvider theme={theme}>
            <CategoryTagList tags={testTags} />
          </ThemeProvider>
        );
        expect(wrapper).toMatchSnapshot();
      });
    });

    test('to contain the correct count of children', () => {
      const wrapper = mount(
        <ThemeProvider theme={theme}>
          <CategoryTagList tags={testTags} />
        </ThemeProvider>
      );
      expect(wrapper.find('CategoryTag')).toHaveLength(4);
    });

    test("when passed the optional prop 'withPreIcon'", () => {
      const wrapper = mount(
        <ThemeProvider theme={theme}>
          <CategoryTagList tags={testTags} withPreIcon />
        </ThemeProvider>
      );
      expect(wrapper.find('S-PreIconWrapperDiv')).toHaveLength(1);
    });

    test("when NOT passed the optional prop 'withPreIcon' (default)", () => {
      const wrapper = mount(
        <ThemeProvider theme={theme}>
          <CategoryTagList tags={testTags} />
        </ThemeProvider>
      );
      expect(wrapper.find('S-PreIconWrapperDiv')).toHaveLength(0);
    });
  });

  describe('has proper style rules', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <CategoryTagList tags={testTags} withPreIcon />
      </ThemeProvider>
    );

    test('in ComponentWrapperDiv', () => {
      expect(wrapper.find('S-ComponentWrapperDiv').first()).toHaveStyleRule(
        'display',
        'flex'
      );
      expect(
        wrapper.find('S-ComponentWrapperDiv').first()
      ).toHaveStyleRule('margin-right', '0.7rem', { target: '*' });
      expect(
        wrapper.find('S-ComponentWrapperDiv').first()
      ).toHaveStyleRule('margin-bottom', '0.7rem', { target: '*' });
    });

    test('in PreIconWrapperDiv', () => {
      expect(wrapper.find('S-PreIconWrapperDiv').first()).toHaveStyleRule(
        'display',
        'flex'
      );
      expect(wrapper.find('S-PreIconWrapperDiv').first()).toHaveStyleRule(
        'align-items',
        'center'
      );
      expect(wrapper.find('S-PreIconWrapperDiv').first()).toHaveStyleRule(
        'color',
        `${theme.color.text.primaryLight}`
      );
    });
  });
});
