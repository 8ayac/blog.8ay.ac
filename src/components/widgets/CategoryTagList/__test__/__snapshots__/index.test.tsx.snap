// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryTagList is rendered correctly to match the snapshot with preIcon 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.emotion-0>* {
  margin-right: 0.7rem;
  margin-bottom: 0.7rem;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsla(0, 0%, 40%);
}

.emotion-4 {
  padding: 0.2rem 1rem;
  font-size: 0.8em;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: hsla(212, 72%, 50%);
  -webkit-filter: drop-shadow(
        2px 2px 1px hsla(212, 72%, 80%)
      );
  filter: drop-shadow(
        2px 2px 1px hsla(212, 72%, 80%)
      );
  border-radius: 3px;
}

.emotion-4:hover {
  color: white;
  background-color: hsla(212, 72%, 60%);
}

.emotion-4:visited {
  color: white;
}

<ThemeProvider
  theme={
    Object {
      "color": Object {
        "blue": Object {
          "base": "hsla(212, 72%, 60%)",
          "dark": "hsla(212, 72%, 50%)",
          "dark2": "hsla(212, 72%, 40%)",
          "dark3": "hsla(212, 72%, 30%)",
          "dark4": "hsla(212, 72%, 20%)",
          "light": "hsla(212, 72%, 70%)",
          "light2": "hsla(212, 72%, 80%)",
          "light3": "hsla(212, 72%, 90%)",
        },
        "border": Object {
          "primary": "hsla(0, 0%, 50%)",
          "primaryLight": "hsla(0, 0%, 75%)",
        },
        "green": Object {
          "base": "hsla(148, 72%, 60%)",
          "dark": "hsla(148, 72%, 50%)",
          "dark2": "hsla(148, 72%, 40%)",
          "dark3": "hsla(148, 72%, 30%)",
          "dark4": "hsla(148, 72%, 20%)",
          "light": "hsla(148, 72%, 70%)",
          "light2": "hsla(148, 72%, 80%)",
          "light3": "hsla(148, 72%, 90%)",
        },
        "primary": "hsla(212, 72%, 60%)",
        "primaryDark": "hsla(212, 72%, 50%)",
        "primaryLight": "hsla(212, 72%, 70%)",
        "text": Object {
          "accent": "hsla(148, 72%, 50%)",
          "primary": "hsla(0, 0%, 15%)",
          "primaryLight": "hsla(0, 0%, 40%)",
          "secondary": "hsla(212, 72%, 40%)",
        },
        "yellow": Object {
          "base": "hsla(60, 72%, 60%)",
          "dark": "hsla(60, 72%, 50%)",
          "dark2": "hsla(60, 72%, 40%)",
          "dark3": "hsla(60, 72%, 30%)",
          "dark4": "hsla(60, 72%, 20%)",
          "light": "hsla(60, 72%, 70%)",
          "light2": "hsla(60, 72%, 80%)",
          "light3": "hsla(60, 72%, 90%)",
        },
      },
    }
  }
>
  <CategoryTagList
    tags={
      Array [
        "A",
        "B",
        "C",
        "DUPLICATE",
        "DUPLICATE",
      ]
    }
    withPreIcon={true}
  >
    <S-ComponentWrapperDiv>
      <div
        className="emotion-0 emotion-1"
      >
        <S-PreIconWrapperDiv>
          <div
            className="emotion-2 emotion-3"
          >
            <FontAwesomeIcon
              border={false}
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    640,
                    512,
                    Array [],
                    "f02c",
                    "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                  ],
                  "iconName": "tags",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-tags fa-w-20 "
                data-icon="tags"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
          </div>
        </S-PreIconWrapperDiv>
        <CategoryTag
          key="A"
          name="A"
        >
          <Link
            href="/tags/A"
            passHref={true}
          >
            <Anchor
              href="/tags/A"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <a
                className="emotion-4 emotion-5"
                href="/tags/A"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                A
              </a>
            </Anchor>
          </Link>
        </CategoryTag>
        <CategoryTag
          key="B"
          name="B"
        >
          <Link
            href="/tags/B"
            passHref={true}
          >
            <Anchor
              href="/tags/B"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <a
                className="emotion-4 emotion-5"
                href="/tags/B"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                B
              </a>
            </Anchor>
          </Link>
        </CategoryTag>
        <CategoryTag
          key="C"
          name="C"
        >
          <Link
            href="/tags/C"
            passHref={true}
          >
            <Anchor
              href="/tags/C"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <a
                className="emotion-4 emotion-5"
                href="/tags/C"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                C
              </a>
            </Anchor>
          </Link>
        </CategoryTag>
        <CategoryTag
          key="DUPLICATE"
          name="DUPLICATE"
        >
          <Link
            href="/tags/DUPLICATE"
            passHref={true}
          >
            <Anchor
              href="/tags/DUPLICATE"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <a
                className="emotion-4 emotion-5"
                href="/tags/DUPLICATE"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                DUPLICATE
              </a>
            </Anchor>
          </Link>
        </CategoryTag>
      </div>
    </S-ComponentWrapperDiv>
  </CategoryTagList>
</ThemeProvider>
`;

exports[`CategoryTagList is rendered correctly to match the snapshot without preIcon (default) 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.emotion-0>* {
  margin-right: 0.7rem;
  margin-bottom: 0.7rem;
}

.emotion-2 {
  padding: 0.2rem 1rem;
  font-size: 0.8em;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: hsla(212, 72%, 50%);
  -webkit-filter: drop-shadow(
        2px 2px 1px hsla(212, 72%, 80%)
      );
  filter: drop-shadow(
        2px 2px 1px hsla(212, 72%, 80%)
      );
  border-radius: 3px;
}

.emotion-2:hover {
  color: white;
  background-color: hsla(212, 72%, 60%);
}

.emotion-2:visited {
  color: white;
}

<ThemeProvider
  theme={
    Object {
      "color": Object {
        "blue": Object {
          "base": "hsla(212, 72%, 60%)",
          "dark": "hsla(212, 72%, 50%)",
          "dark2": "hsla(212, 72%, 40%)",
          "dark3": "hsla(212, 72%, 30%)",
          "dark4": "hsla(212, 72%, 20%)",
          "light": "hsla(212, 72%, 70%)",
          "light2": "hsla(212, 72%, 80%)",
          "light3": "hsla(212, 72%, 90%)",
        },
        "border": Object {
          "primary": "hsla(0, 0%, 50%)",
          "primaryLight": "hsla(0, 0%, 75%)",
        },
        "green": Object {
          "base": "hsla(148, 72%, 60%)",
          "dark": "hsla(148, 72%, 50%)",
          "dark2": "hsla(148, 72%, 40%)",
          "dark3": "hsla(148, 72%, 30%)",
          "dark4": "hsla(148, 72%, 20%)",
          "light": "hsla(148, 72%, 70%)",
          "light2": "hsla(148, 72%, 80%)",
          "light3": "hsla(148, 72%, 90%)",
        },
        "primary": "hsla(212, 72%, 60%)",
        "primaryDark": "hsla(212, 72%, 50%)",
        "primaryLight": "hsla(212, 72%, 70%)",
        "text": Object {
          "accent": "hsla(148, 72%, 50%)",
          "primary": "hsla(0, 0%, 15%)",
          "primaryLight": "hsla(0, 0%, 40%)",
          "secondary": "hsla(212, 72%, 40%)",
        },
        "yellow": Object {
          "base": "hsla(60, 72%, 60%)",
          "dark": "hsla(60, 72%, 50%)",
          "dark2": "hsla(60, 72%, 40%)",
          "dark3": "hsla(60, 72%, 30%)",
          "dark4": "hsla(60, 72%, 20%)",
          "light": "hsla(60, 72%, 70%)",
          "light2": "hsla(60, 72%, 80%)",
          "light3": "hsla(60, 72%, 90%)",
        },
      },
    }
  }
>
  <CategoryTagList
    tags={
      Array [
        "A",
        "B",
        "C",
        "DUPLICATE",
        "DUPLICATE",
      ]
    }
  >
    <S-ComponentWrapperDiv>
      <div
        className="emotion-0 emotion-1"
      >
        <CategoryTag
          key="A"
          name="A"
        >
          <Link
            href="/tags/A"
            passHref={true}
          >
            <Anchor
              href="/tags/A"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <a
                className="emotion-2 emotion-3"
                href="/tags/A"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                A
              </a>
            </Anchor>
          </Link>
        </CategoryTag>
        <CategoryTag
          key="B"
          name="B"
        >
          <Link
            href="/tags/B"
            passHref={true}
          >
            <Anchor
              href="/tags/B"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <a
                className="emotion-2 emotion-3"
                href="/tags/B"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                B
              </a>
            </Anchor>
          </Link>
        </CategoryTag>
        <CategoryTag
          key="C"
          name="C"
        >
          <Link
            href="/tags/C"
            passHref={true}
          >
            <Anchor
              href="/tags/C"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <a
                className="emotion-2 emotion-3"
                href="/tags/C"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                C
              </a>
            </Anchor>
          </Link>
        </CategoryTag>
        <CategoryTag
          key="DUPLICATE"
          name="DUPLICATE"
        >
          <Link
            href="/tags/DUPLICATE"
            passHref={true}
          >
            <Anchor
              href="/tags/DUPLICATE"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <a
                className="emotion-2 emotion-3"
                href="/tags/DUPLICATE"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                DUPLICATE
              </a>
            </Anchor>
          </Link>
        </CategoryTag>
      </div>
    </S-ComponentWrapperDiv>
  </CategoryTagList>
</ThemeProvider>
`;
